File: app/src/main/java/sherafatpour/entekhab/notify/AddEditeNoteActivity.java
package sherafatpour.entekhab.notify;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.NumberPicker;
import android.widget.Toast;

public class AddEditeNoteActivity extends AppCompatActivity {
    public static final String EXTRA_TITLE =
            "sherafatpour.entekhab.notify.EXREA_TITLE";
    public static final String EXTRA_DESCRIPTION =
            "sherafatpour.entekhab.notify.EXTRA_DESCRIPTION";
    public static final String EXTRA_PRIORITY =
            "sherafatpour.entekhab.notify.EXTRA_PRIORITY";
    public static final String EXTRA_ID =
            "sherafatpour.entekhab.notify.EXTRA_ID";
    private EditText editTextTitle;
    private EditText editTextDescription;
    private NumberPicker numberPickerPriority;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_note);

        editTextTitle = findViewById(R.id.edt_text_title);
        editTextDescription = findViewById(R.id.edt_text_description);
        numberPickerPriority = findViewById(R.id.number_picker_priority);

        numberPickerPriority.setMinValue(1);
        numberPickerPriority.setMaxValue(10);

        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_close);
        Intent intent = getIntent();

        if (intent.hasExtra(EXTRA_ID)){
            setTitle("Edit Notification");
            editTextTitle.setText(intent.getStringExtra(EXTRA_TITLE));
            editTextDescription.setText(intent.getStringExtra(EXTRA_DESCRIPTION));
            numberPickerPriority.setValue(intent.getIntExtra(EXTRA_PRIORITY,1));

        }else{
            setTitle("Add Notification");

        }

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.menu, menu);
        return true;
    }


    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()) {
            case R.id.save_note:
                savetNote();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void savetNote() {

        String title = editTextTitle.getText().toString();
        String description = editTextDescription.getText().toString();
        int priority = numberPickerPriority.getValue();

        if (title.trim().isEmpty() || description.trim().isEmpty()) {
            Toast.makeText(this, "Please insert a title and description", Toast.LENGTH_SHORT).show();
        }

        Intent data = new Intent();
        data.putExtra(EXTRA_TITLE, title);
        data.putExtra(EXTRA_DESCRIPTION, description);
        data.putExtra(EXTRA_PRIORITY, priority);
        int id = getIntent().getIntExtra(EXTRA_ID,-1);
        if (id !=-1){
            data.putExtra(EXTRA_ID,id);

        }
        setResult(RESULT_OK, data);
        finish();
    }
}

File: app/src/main/java/sherafatpour/entekhab/notify/view/AddEditeNoteActivity.kt
package sherafatpour.entekhab.notify.view

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.EditText
import android.widget.NumberPicker
import android.widget.Toast
import sherafatpour.entekhab.notify.R

class AddEditeNoteActivity : AppCompatActivity() {
    private var editTextTitle: EditText? = null
    private var editTextDescription: EditText? = null
    private var numberPickerPriority: NumberPicker? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_note)

        editTextTitle = findViewById(R.id.edt_text_title)
        editTextDescription = findViewById(R.id.edt_text_description)
        numberPickerPriority = findViewById(R.id.number_picker_priority)

        numberPickerPriority!!.minValue = 1
        numberPickerPriority!!.maxValue = 10

        supportActionBar!!.setHomeAsUpIndicator(R.drawable.ic_close)
        val intent = intent

        if (intent.hasExtra(EXTRA_ID)) {
            title = "Edit Notification"
            editTextTitle!!.setText(intent.getStringExtra(EXTRA_TITLE))
            editTextDescription!!.setText(intent.getStringExtra(EXTRA_DESCRIPTION))
            numberPickerPriority!!.value = intent.getIntExtra(EXTRA_PRIORITY, 1)

        } else {
            title = "Add Notification"

        }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val menuInflater = menuInflater
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
            R.id.save_note -> {
                saveNote()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    private fun saveNote() {

        val title = editTextTitle!!.text.toString()
        val description = editTextDescription!!.text.toString()
        val priority = numberPickerPriority!!.value

        if (title.trim { it <= ' ' }.isEmpty() || description.trim { it <= ' ' }.isEmpty()) {
            Toast.makeText(this, "Please insert a title and description", Toast.LENGTH_SHORT).show()
        }
 //send params to previous activity by intent
        val data = Intent()
        data.putExtra(EXTRA_TITLE, title)
        data.putExtra(EXTRA_DESCRIPTION, description)
        data.putExtra(EXTRA_PRIORITY, priority)
        val id = intent.getIntExtra(EXTRA_ID, -1)
        if (id != -1) {
            data.putExtra(EXTRA_ID, id)

        }
        setResult(Activity.RESULT_OK, data)
        finish()
    }

    companion object {
        val EXTRA_TITLE = "sherafatpour.entekhab.notify.EXREA_TITLE"
        val EXTRA_DESCRIPTION = "sherafatpour.entekhab.notify.EXTRA_DESCRIPTION"
        val EXTRA_PRIORITY = "sherafatpour.entekhab.notify.EXTRA_PRIORITY"
        val EXTRA_ID = "sherafatpour.entekhab.notify.EXTRA_ID"
    }
}
